extensions/logging/README.md

```diff
@@ -0,0 +1,158 @@
+```
+
+In the constructor, add the component to your application:
+
+```ts
+this.component(LoggingComponent);
+```
+
+Now your application can add a controller as follows to leverage the logging
+facilities:
+
+```ts
+import {inject} from '@loopback/context';
+import {Logger} from '@loopback/extension-logging';
+
+class MyController {
+  // Inject a winston logger
+  @inject(LoggingBindings.WINSTON_LOGGER)
+  private logger: Logger;
+
+  // http access is logged by a global interceptor
+  @get('/greet/{name}')
+  // log the `greet` method invocations
+  @log()
+  greet(@param.path.string('name') name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @get('/hello/{name}')
+  hello(@param.path.string('name') name: string) {
+    // Use the winston logger explicitly
+    this.logger.log('info', `greeting ${name}`);
+    return `Hello, ${name}`;
+  }
+}
+```
+
+## Configure the logging component
+
+The component contributes bindings with keys declared in `LoggingBindings`
+namespace below:
+
+- FLUENT_SENDER - A fluent sender
+- WINSTON_LOGGER - A winston logger
+- WINSTON_TRANSPORT_FLUENT - A fluent transport for winston
+- WINSTON_INTERCEPTOR - A local interceptor set by `@log` to log method
+  invocations
+- WINSTON_ACCESS_LOGGER - A global interceptor that logs http access with
+  [Morgan](https://github.com/expressjs/morgan) format
+
+The fluent sender and transport for winston can be configured against
+`FLUENT_SENDER`:
+
+```ts
+import {LoggingBindings} from '@loopback/extension-logging';
+
+app.configure(LoggingBindings.FLUENT_SENDER).to({
+  host: process.env.FLUENTD_SERVICE_HOST || 'localhost',
+  port: +(process.env.FLUENTD_SERVICE_PORT_TCP || 0) || 24224,
+  timeout: 3.0,
+  reconnectInterval: 600000, // 10 minutes
+});
+```
+
+The winston logger can be configured against `LoggingBindings.WINSTON_LOGGER`:
+
+```ts
+import {LoggingBindings} from '@loopback/extension-logging';
+
+ctx.configure<LoggerOptions>(LoggingBindings.WINSTON_LOGGER).to({
+  level: 'info',
+  format: format.json(),
+  defaultMeta: {framework: 'LoopBack'},
+});
+```
+
+The winston logger accepts two types of extensions to the following extension
+points:
+
+- WINSTON_TRANSPORT = 'logging.winston.transport'
+- WINSTON_FORMAT = 'logging.winston.format'
+
+```ts
+import {extensionFor} from '@loopback/core';
+import {format} from 'winston';
+import {WINSTON_FORMAT, WINSTON_TRANSPORT} from '@loopback/extension-logging';
+
+const myFormat: Format = ...;
+
+ctx
+  .bind('logging.winston.formats.myFormat')
+  .to(myFormat)
+  .apply(extensionFor(WINSTON_FORMAT));
+ctx
+  .bind('logging.winston.formats.colorize')
+  .to(format.colorize())
+  .apply(extensionFor(WINSTON_FORMAT));
+```
+
+The access log interceptor can also be configured to customize
+[Morgan format and options](https://github.com/expressjs/morgan#morganformat-options):
+
+```
+ctx.configure(LoggingBindings.WINSTON_ACCESS_LOGGER).to({format: 'combined'});
+```
```

Comment:
How is this interceptor applied? IIUC, it's defined as a global interceptor, but IIRC our boot does not load global interceptors from the extensions. If it's automatically registered by other means, then how can the application developer disable this logger?

Reply:
The component registers the interceptor. It can be disabled by setting `enableHttpAccessLog` to `false` for the component config.